wallet/datastore.go

```diff
@@ -969,3 +987,25 @@ func createTx(ctx context.Context, pg *Postgres) (tx *sqlx.Tx, err error) {
 	}
 	return tx, nil
 }
+
+func waitAndLock(ctx context.Context, tx *sqlx.Tx, id uuid.UUID) error {
```

Comment:
If this is to be generic, suggest passing in param as an `v interface{}` instead of `id uuid.UUID`.  If it is just to be for application level locking of uuids, then this is fine i suppose?

Reply:
Suggest using the same getTx model from the context that we do elsewhere in wallets instead of passing in ctx and tx individually.  This way if no tx is passed in, that is fine, it will create a tx.  If there is a Tx on the context, it will use that.  Here is an example: 
https://github.com/brave-intl/bat-go/blob/master/wallet/datastore.go#L384-L390

Reply:
changed to use  getTx